7_react/react1/week3/my-app/src/AddTodo.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+const AddTodo = ({ todos, setTodos }) => {
+  const [description, setDescription] = useState("");
+  const [deadline, setDeadline] = useState("");
+
+  const onChangeDescription = (e) => {
+    setDescription(e.target.value);
+  };
+
+  const onChangeDeadline = (e) => {
+    setDeadline(e.target.value);
+  };
+
+  return (
+    <div>
+      Todo description:
+      <input type="text" value={description} onChange={onChangeDescription} />
+      <br />
+      Deadline:
+      <input type="date" value={deadline} onChange={onChangeDeadline} />
+      <button
+        onClick={() =>
+          setTodos([
+            ...todos,
+            {
+              id: uuidv4(),
+              description: description,

```

Comment:
A minor thing: in JavaScript, when the two sides of an assignment have the same name, you can just write it once, i.e. `description, `.